lib/modules/manager/puppet/puppetfile-parser.ts

```diff
@@ -0,0 +1,106 @@
+import { newlineRegex, regEx } from '../../../util/regex';
+import type { PuppetForgeUrl, PuppetfileModule } from './types';
+
+const forgeRegex = regEx(/^forge\s+['"]([^'"]+)['"]/);
+const commentRegex = regEx(/#.*$/);
+
+/**
+ * For us a Puppetfile is build up of forges that have Modules.
+ *
+ * Modules are the updatable parts.
+ *
+ */
+export class Puppetfile {
+  private forgeModules: Map<PuppetForgeUrl, PuppetfileModule[]> = new Map<
```

Comment:
```suggestion
  private readonly forgeModules = new Map<
```
no need for duplicate type definition